modin/pandas/test/dataframe/test_indexing.py

```diff
@@ -412,6 +412,31 @@ def test_loc(data):
     with pytest.raises(KeyError):
         modin_df.loc["NO_EXIST"]
 
+    # From issue #3764
```

Comment:
All the tests you're adding should go in a separate test case with a name like `test_insert_with_loc`. This test case is already way too long.

Reply:
Done.